src/Avalonia.Build.Tasks/XamlCompilerTaskExecutor.cs

```diff
@@ -370,15 +403,11 @@ bool CompileGroup(IResourceGroup group)
                                     if (i[c].OpCode == OpCodes.Call)
                                     {
                                         var op = i[c].Operand as MethodReference;
-                                        
-                                        // TODO: Throw an error
-                                        // This usually happens when the same XAML resource was added twice for some weird reason
-                                        // We currently support it for dual-named default theme resources
                                         if (op != null
                                             && op.Name == TrampolineName)
                                         {
-                                            foundXamlLoader = true;
-                                            break;
+                                            throw new InvalidProgramException("Same XAML file was loaded twice." +
+                                                "Make sure there is no x:Class duplicates no files were added to the AvaloniaResource msbuild items group twice.");
```

Comment:
@kekekeks as I understand, we don't need this hack anymore.

Reply:
IIRC it was required for compatibility with pre-`.axaml` themes that was achieved via including the same file as EmbeddedResource and AvaloniaResource.